dll_tests.js

```diff
@@ -0,0 +1,323 @@
+            assertEqual(dLinkedList.tail, dLinkedList.head);
+            
+            console.log('deleteAt OK');
+        },
+        
+        deleteNodeTest: function () {
+            
+            var dLinkedList = DLinkedList();
+            
+            assertError.call(dLinkedList,'this.deleteNode(0);','The specified node was not found');
+            
+            
+            for (var i = 0; i < 3; i++) {
+                
+                dLinkedList.insertRight(i);
+            }
+            
+            assertError.call(dLinkedList,'this.deleteNode(5);','The specified node was not found');
+            
+            dLinkedList.deleteNode(1);
+            assertEqual(dLinkedList.head.data, 0);
+            assertEqual(dLinkedList.tail.data, 2);
+            assertEqual(dLinkedList.head.next.data, dLinkedList.tail.data);
+            
+            dLinkedList.deleteNode(2);
+            assertEqual(dLinkedList.head.data, 0);
+            assertEqual(dLinkedList.head.data, dLinkedList.tail.data);
+            
+            
+            assertError.call(dLinkedList,'this.deleteNode(2);','The specified node was not found');
+            
+            dLinkedList.deleteNode(0);
+            assertEqual(dLinkedList.head, null);
+            assertEqual(dLinkedList.tail, dLinkedList.head);
+            
+            console.log('deleteNode OK');
+        },
+        
+        displayForwardTest: function () {
+            
+            var dLinkedList = DLinkedList();
+            
+            for (var i = 0; i < 3; i++) {
+                
+                dLinkedList.insertRight(i);
+            }
+            
+            assertEqual(dLinkedList.displayForward(), '0,1,2')
+        },
+        
+        displayBackwardTest: function () {
+            
+            var dLinkedList = DLinkedList();
+            
+            for (var i = 0; i < 3; i++) {
+                
+                dLinkedList.insertRight(i);
+            }
+            
+            assertEqual(dLinkedList.displayBackward(), '2,1,0')
+        },
+        
+        getSizeTest: function () {
+            
+            var dLinkedList = DLinkedList();
+            
+            for (var i = 0; i < 3; i++) {
+                
+                dLinkedList.insertRight(i);
+            }
+            
+            assertEqual(dLinkedList.getSize(), 3);
+        },
+        
+        clearTest: function () {
+            
+            var dLinkedList = DLinkedList();
+            
+            for (var i = 0; i < 3; i++) {
+                
+                dLinkedList.insertRight(i);
+            }
+            
+            dLinkedList.clear();
+            assertEqual(dLinkedList.head, null);
+            assertEqual(dLinkedList.tail, dLinkedList.head);
+        }
+    }
+    
+    var unitTestsMethods = Object.keys(unitTests);
+    
+    for (var i = 0; i < unitTestsMethods.length; i++) {
+        
+         unitTests[unitTestsMethods[i]]();
```

Comment:
You should try catch it to log errors but potentially continue running other tests instead of crashing the program.